lib/membrane_matroska/serializer/matroska.ex

```diff
@@ -180,11 +172,21 @@ defmodule Membrane.Matroska.Serializer.Matroska do
        TrackType: 1,
        CodecID: "V_MPEG4/ISO/AVC",
        FlagLacing: 0,
-       TrackUID: id,
+       TrackUID: get_track_uid(id),
        TrackNumber: track_number
      ]}
   end
 
+  defp get_track_uid(id) when is_reference(id) do
```

Comment:
I think you need to handle a few more types here
```suggestion
  defp get_track_uid(id) when not is_integer(id) do
```

Btw, sometimes it's good practice to avoid negated guards, so you might also switch the clauses, but I don't insist